disk-encryption-tool

```diff
@@ -221,9 +233,67 @@ make_ro()
 	btrfs prop set -t s "$mp" ro true
 }
 
+enroll_fido2()
+{
+	if [ ! -e /usr/bin/systemd-cryptenroll ]; then
+		log_info "missing systemd-cryptenroll in initrd"
+		return
+	fi
+
+	devices="$(systemd-cryptenroll --fido2-device=list 2>&1)"
+	if grep -q "No FIDO2" <<<"$devices"; then
+		echo "No FIDO2 devices found"
+		return
+	fi
+
+	echo -e "\e[1mFIDO2 device present. If registered will take precedence from TPM2\e[m"
+
+	umask 066; echo -n "$password" > "$tmpdir"/pw; umask 022
+	systemd-cryptenroll --unlock-key-file="$tmpdir"/pw --fido2-device=auto "$blkpart" && with_fido2=1
```

Comment:
would it be possible to use a fd here? ie < <(echo "$password")
Longer term I think we need to teach systemd-cryptenroll to read the passphrase using different means.

Reply:
I do not think so. The only two options are using a named file (there is a fopen in the code), or a env var (that is advised against in the notes[1])

Do you want me to go to the `PASSWORD=` path?

[1] https://systemd.io/ENVIRONMENT/

Reply:
`< <(...)`  is process substitution. It produces a path argument that can be opened. Programs that just read from a file should work. Fails if the program wants to seek though.

Reply:
oh let me try

Reply:
pardon. only one `<, otherwise it would redirect stdin. Anyway I found that cryptenroll also reads from kernel keyring. So I could store the generated key there and cryptenroll would pick it up automatically.

Reply:
I am moving this out, so the new PR will take some time